design-documents/database/declarative-schema-database-compatibility.md

```diff
@@ -0,0 +1,1093 @@
+{
+    public const TYPE = 'VARCHAR';
+
+    /**
+     * Returns the maximum length in characters.
+     *
+     * @return int
+     */
+    public function getCharLength(): int {}
+
+    /**
+     * Returns the character set name.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
+     *
+     * @return string|null
+     */
+    public function getCharsetName(): ?string {}
+
+    /**
+     * Returns the collation name.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
+     *
+     * @return string|null
+     */
+    public function getCollationName(): ?string {}
+
+    /**
+     * Returns the default value of the column.
+     *
+     * @return string|null
+     */
+    public function getDefaultValue(): ?string {}
+}
+```
+
+```
+class InformationSchema\Table\NationalVarcharColumn extends Column
+{
+    public const TYPE = 'NATIONAL VARCHAR';
+
+    /**
+     * Returns the maximum length in characters.
+     *
+     * @return int
+     */
+    public function getCharLength(): int {}
+
+    /**
+     * Returns the character set name.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
+     *
+     * @return string|null
+     */
+    public function getCharsetName(): ?string {}
+
+    /**
+     * Returns the collation name.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
+     *
+     * @return string|null
+     */
+    public function getCollationName(): ?string {}
+
+    /**
+     * Returns the default value of the column.
+     *
+     * @return string|null
+     */
+    public function getDefaultValue(): ?string {}
+}
+```
+Note that MySQL / MariaDB maps this type to `varchar`.
+
+```
+class InformationSchema\Table\ClobColumn extends Column
+{
+    public const TYPE = 'CLOB';
+
+    /**
+     * Returns the character set name.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
+     *
+     * @return string|null
+     */
+    public function getCharsetName(): ?string {}
+
+    /**
+     * Returns the collation name.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
+     *
+     * @return string|null
+     */
+    public function getCollationName(): ?string {}
+
+    /**
+     * Returns the default value of the column.
+     *
+     * @return string|null
+     */
+    public function getDefaultValue(): ?string {}
+}
+```
+Note that MySQL / MariaDB and PostgreSQL use the type `text` for this purpose.
+
+```
+class InformationSchema\Table\NationalClobColumn extends Column
+{
+    public const TYPE = 'NCLOB';
+
+    /**
+     * Returns the character set name.
+     *
+     * Note that not all RDBMS support this, returning `null` indicates whether this is supported.
```

Comment:
Then we can't make type like this.

Reply:
@melnikovi Do you suggest to drop all non-ANSI-SQL features from column classes or do you have something else in mind?

Reply:
Yes, let's limit column types to currently used (can check declarative schema to see what is used).